### PR TITLE
Supporting Rancher RKE

### LINKED ISSUE #771: supporting Rancher RKE
## Feature Request

Kubearmor has not be tested with [Rancher RKE](https://rancher.com/products/rke). The task involves setting up the sample cluster, verifying kubearmor operations on the cluster and making any fixes if necessary.

## Tasks
- [ ] setting up a sample cluster using Rancher RKE
- [ ] installing kubearmor on the sample cluster
- [ ] execute the automated tests on the sample cluster
- [ ] if kubearmor does not work, diagnose the issue
- [ ] discuss with the community and fix the issue
- [ ] update [kubearmor support matrix](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/support_matrix.md)

### PR DESCRIPTION
**Pull request by PrimalPimmy**
Fixes #771 .

Not filling up the support matrix till all tests are passed.

### CODE REVIEW

File: deployments/RKE/kubearmor.yaml [hunk 1]
```diff
@@ -0,0 +1,1186 @@
...
+          type: Directory
+        name: sys-kernel-debug-path
+      - hostPath:
+          path: /etc/os-release
+          type: File
+        name: os-release-path
+      - hostPath:
+          path: /usr/src
+          type: Directory
+        name: usr-src-path
```
Comment (nyrahul) [inline deployments/RKE/kubearmor.yaml:192]: I thought the whole problem with supporting RKE was to use `DirectoryOrCreate` here. Isn't it?
```suggestion
      - hostPath:
          path: /usr/src
          type: DirectoryOrCreate
        name: usr-src-path
```

**Review by nyrahul — CHANGES_REQUESTED** [review]
Hey @PrimalPimmy , can you test whether the manifest generated in this PR can be used to install on RKE? Based on what I see, it won't work.

File: deployments/RKE/kubearmor.yaml (hunk 1 above)
Comment (PrimalPimmy) [reply deployments/RKE/kubearmor.yaml:192]: I thought it only applied to /lib/modules, and not /usr/src?

Comment (PrimalPimmy) [reply deployments/RKE/kubearmor.yaml:192]: we only changed it to DirectoryOrCreate only for /lib/modules path as far as I remember. rest were working fine.

**Comment by PrimalPimmy** [comment]
I have confirmed that RKE works with this configuration. Anyone else can test it if you want.

**Comment by achrefbensaad** [comment]
@nyrahul @PrimalPimmy Ill re-test this PR.

**Review by achrefbensaad — CHANGES_REQUESTED** [review]
This manifest does not work on rke.
I tried installing using kamor install (it worked)
I tried using the generic manifest and it worked also.

image

**Comment by PrimalPimmy** [comment]
@achrefbensaad I'm looking into it.. btw, this manifest is generated when we do karmor install --save, right?

**Comment by achrefbensaad** [comment]
No, We generate it by running, `make gen` under `deployments`

**Comment by PrimalPimmy** [comment]
I did that after editing main.go to add RKE. Perhaps because I didn't make changes to defaults,go, is that why it's not working?

**Comment by achrefbensaad** [comment]
are there any changes that require updating `default.go` ?

**Comment by PrimalPimmy** [comment]
I changed 
image

from Directory to DirectoryOrCreate. 

But now for some reason, it even works without this change. Before it would get stuck on init if this change is not done

**Comment by daemon1024** [comment]
> But now for some reason, it even works without this change.

Because your first installation created the Directory ( due to OrCreate ), so for subsequent installs it's available.

**Comment by daemon1024** [comment]
If we are needing DirectoryOrCreate, that means it's not serving any purpose right? It should fine to just not mount it then? Can we try that?

**Comment by PrimalPimmy** [comment]
hey @achrefbensaad , can you test this in a new rke environment? did the changes, should work now hopefully

**Comment by nyrahul** [comment]
> hey @achrefbensaad , can you test this in a new rke environment? did the changes, should work now hopefully

Thanks @achrefbensaad for supporting this ticket with such rigor.

**Comment by achrefbensaad** [comment]
We opted for directory or create for all plateforms. https://github.com/kubearmor/KubeArmor/pull/872

**Pull request closed without merging**